tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/ProjectSystem/UpToDate/BuildUpToDateCheckTests.cs

```diff
@@ -1093,7 +1093,40 @@ public async Task IsUpToDateAsync_False_CopyToOutputDirectoryDestinationDoesNotE
         }
 
         [Fact]
-        public async Task IsUpToDateAsync_False_AdditionalDependentFileTimeSourceIsNewerThanOutput()
+        public async Task IsUpToDateAsync_False_AdditionalDependentFileTimeSourceRemoved()
+        {
+            var projectSnapshot = new Dictionary<string, IProjectRuleSnapshotModel>
+            {
+                [UpToDateCheckBuilt.SchemaName] = SimpleItems("Output")
+            };
+
+            var itemChangeTime = DateTime.UtcNow.AddMinutes(-4);
+            var lastCheckTime = DateTime.UtcNow.AddMinutes(-3);
+            var dependentTime = DateTime.UtcNow.AddMinutes(-2);
+            var outputTime = DateTime.UtcNow.AddMinutes(-1);
+
+            var dependentPath = @"C:\Dev\Solution\Project\Dependent";
+            var outputPath = @"C:\Dev\Solution\Project\Output";
+
+            await SetupAsync(
+                projectSnapshot: projectSnapshot,
+                dependentTimeFiles: new[] { (dependentPath, dependentTime) });
+
+            _fileSystem.AddFile(dependentPath, dependentTime);
+            _fileSystem.AddFile(outputPath, outputTime);
+            _buildUpToDateCheck.TestAccess.SetLastCheckedAtUtc(lastCheckTime);
+            _buildUpToDateCheck.TestAccess.SetLastItemsChangedAtUtc(itemChangeTime);
+
+            // Remove dependentPath
+            await SetupAsync(projectSnapshot: projectSnapshot);
+
+            await AssertNotUpToDateAsync(
+                "Some additional dependent files are added or removed, not up to date.",
+                telemetryReason: "AdditionalDependentFiles");
```

Comment:
The message should be something like:

> The set of AdditionalDependentFiles was changed more recently ({localChangeTime}) than the earliest output '{outputPath}' ({localOutputTime}), not up to date.

And the telemetry reason should probably be `"Outputs"` to match other instances.